tests/test_ui.py

```diff
@@ -635,3 +549,121 @@ def test_position_search_modal(selenium, app, status, warning):
+
+
+@pytest.mark.sphinx(srcdir=TEST_DOCS_SRC)
+def test_writing_query_adds_rtd_search_as_url_param(selenium, app, status, warning):
+    """Test if the `rtd_search` query param is added to the url when user is searching."""
+    app.build()
+    path = app.outdir / 'index.html'
+
+    # to test this, we need to override the $.ajax function
+    ajax_func = get_ajax_overwrite_func('error')
+    injected_script = SCRIPT_TAG + ajax_func
+
+    with InjectJsManager(path, injected_script) as _:
+        selenium.get(f'file://{path}')
+        open_search_modal(selenium)
+        query = 'i am searching'
+        query_len = len(query)
+
+        assert (
+            'rtd_search=' not in parse.unquote(selenium.current_url)
+        ), 'rtd_search param must not be present in the url when page loads'
+
+        search_outer_input = selenium.find_element_by_class_name(
+            'search__outer__input'
+        )
+        search_outer_input.send_keys(query)
+        query_param = f'rtd_search={query}'
+
+        assert (
+            query_param in parse.unquote(selenium.current_url)
+        ), 'query param must be present in the url'
+
+        # deleting query from input field
+        for i in range(query_len):
+            search_outer_input.send_keys(Keys.BACK_SPACE)
+
+            if i != query_len -1:
+                
+                current_query = query[:query_len - i - 1]
+                current_url = parse.unquote(selenium.current_url)
+                query_in_url = current_url[current_url.find('rtd_search'):]
+
+                assert (
+                    f'rtd_search={current_query}' == query_in_url
+                )
+
+        assert (
+            'rtd_search=' not in parse.unquote(selenium.current_url)
+        ), 'rtd_search param must not be present in the url if query is empty'
+
+
+@pytest.mark.sphinx(srcdir=TEST_DOCS_SRC)
+def test_modal_open_if_rtd_search_is_present(selenium, app, status, warning):
+    """Test if search modal opens if `rtd_search` query param is present in the URL."""
+    app.build()
+    path = app.outdir / 'index.html'
+
+    # to test this, we need to override the $.ajax function
+    ajax_func = get_ajax_overwrite_func('error')
+    injected_script = SCRIPT_TAG + ajax_func
+
+    with InjectJsManager(path, injected_script) as _:
+        selenium.get(f'file://{path}?rtd_search=i am searching')
+        time.sleep(3)  # give time to open modal and start searching
+
+        search_outer_wrapper = selenium.find_element_by_class_name(
+            'search__outer__wrapper'
+        )
+        search_result_box = selenium.find_element_by_class_name(
+            'search__result__box'
+        )
+
+        assert (
+            search_outer_wrapper.is_displayed() is True
+        ), 'search modal should displayed when the page loads'
+        assert (
+            search_result_box.text == 'Error Occurred. Please try again.'
```

Comment:
Why does this show an error?

Reply:
Because the ajax func is overwritten to return an error here - https://github.com/readthedocs/readthedocs-sphinx-search/pull/25/files#diff-42e9264de947b8999fda83bf2bacd290R610

Reply:
Is there a reason we're testing for an error when we're just trying to see if it shows up?

Reply:
I guess we have to test for something :)

Reply:
There's no particual reason -- I just wanted to test that modal shouldn't be empty after page is loaded and modal is shown.
It also proves that a jquery ajax request was there.